File: src/api/ShowNTell.Domain/Models/Comment.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace ShowNTell.Domain.Models
{
    /// <summary>
    /// A model for an image post comment.
    /// </summary>
    public class Comment
    {
        public int Id { get; set; }
        public string UserEmail { get; set; }
        public int ImagePostId { get; set; }
        public string Content { get; set; }
        public DateTime DateCreated { get; set; }

        public User User { get; set; }
        public ImagePost ImagePost { get;set; }
    }
}

File: src/show-n-tell-vue/src/models/comment.js
function Comment(id, userEmail, username, content, dateCreated){
    this.id = id
    this.userEmail = userEmail
    this.username = username
    this.content = content
    this.dateCreated = new Date(dateCreated)
}

Comment.fromJSON = function(comment) {
    return new Comment(
        comment.id,
        comment.userEmail,
        comment.username,
        comment.content,
        comment.dateCreated
    )
}

export default Comment